src/plugins/ui_stateful/clipboard.ts

```diff
@@ -87,17 +87,16 @@ export class ClipboardPlugin extends UIPlugin {
         const zones = ("target" in cmd && cmd.target) || this.getters.getSelectedZones();
         this.state = this.getClipboardState(zones, cmd.type);
         this.status = "visible";
+        this.allowPaste = true;
         break;
       case "PASTE":
-        if (!this.state) {
+        if (!this.state || !this.allowPaste) {
           break;
         }
+        if (this.state?.operation === "CUT") this.allowPaste = false;
```

Comment:
```suggestion
        if (this.state?.operation === "CUT") {
            this.allowPaste = false;
        }
```
some peoples in the teeam don't like one-line condition, so try to avoid them.